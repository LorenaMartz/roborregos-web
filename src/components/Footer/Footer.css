.footer-row {
  color: rgb(250, 250, 250);
  background-color:  rgb(10, 15, 20);
}

.col-logo {
  padding: 0;
}

.footer-logo {
  width: 60vh;
  height: 31vh;
  object-fit: cover;
  object-position: top;
}

.sitemap-container {
  background-color:  rgb(10, 15, 20);
}

.sitemap-link {
  display: inline-block;
  padding-left: 2.8vw;
  padding-top: 24px;
  color: rgb(250, 250, 250);
  font-size: 15px;
}

.sitemap-link:hover {
  text-decoration: none;
  color: rgb(0, 225, 250);
}

.left-panel {
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  justify-content: space-between;
}

.goback-button {
  padding: 24px 12px;
}

.goback-text {
  font-size: 15px;
}

.row-socialMedia {
  display: flex;
  justify-content: space-evenly;
}

.icon-link {
  color: rgb(250, 250, 250);
  margin: 0 5px;
}

.icon-link:hover {
  color: rgb(0, 225, 250);
}

.mark-text {
  text-align: right;
}

@media (max-width: 992px) {
  .footer-logo {
    width: 100%;
    height: 100%;
    object-position: 100% 20%;
  }

  .sitemap-container {
    text-align: right;
  }

  .sitemap-link {
    padding: 0;
    font-size: 2.5vh;
    min-width: 51%;
  }

  .left-panel {
    justify-content: flex-start;
  }

  .goback-button {
    padding: 10px 0;
  }

  .goback-text {
    font-size: 4vw;
  }

  .row-socialMedia {
    margin: 10px 0;
  }

  .mark-text {
    font-size: 2vh;
    color: #555652;
  }
}
