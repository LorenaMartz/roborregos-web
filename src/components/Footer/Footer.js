import React, { Component } from 'react';
import { Container, Row, Col } from 'react-bootstrap';

class Footer extends Component {
	render() {
		return(
			<div className='footer-container'>
				<Container>
					<Row className='justify-content-md-center'>
						<Col xs='10' md='6'>
							<h5>
								Columna 1
							</h5>
							<p>
								Contenido de columna 1
							</p>
						</Col>
						<Col xs='10' md='6'>
							<h5>
								Columna 2
							</h5>
							<p>
								Contenido de columna 2
							</p>
						</Col>
					</Row>
				</Container>
			</div>
		);
	}
}

export default Footer;