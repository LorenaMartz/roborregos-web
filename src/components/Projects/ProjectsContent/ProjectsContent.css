.projects-content {
    background-color: black;
}