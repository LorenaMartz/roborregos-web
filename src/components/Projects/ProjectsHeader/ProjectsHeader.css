.projects-header {
  height: 100%;
  position: relative;
  background: linear-gradient( 45deg, rgba(20, 18, 19, 0.9) 10%, rgba(0, 25, 105, 0.9) 100%), url(../../../images/github_pattern.svg) top left/auto repeat fixed;
  color: rgba(255, 255, 255, 0.8);
}

.projects-header .main-text-projects {
  color: rgba(255, 255, 255, 0.9);
  font-size: 3vh;
  text-transform: none;
  padding-top: 2%;
  padding-bottom: 0%;
}

.title-text-banner {
  color: rgba(255, 255, 255, 0.9);
  font-size: 7vh;
  text-transform: none;
}

.projects-header .projects-header-footer {
  font-size: 2.5vh;
  text-transform: none;
  display: flex;
  flex-direction: column;
  align-items: center;
  text-align: center;
  position: absolute;
  bottom: 5vh;
  left: 0;
  right: 0;
  margin-left:auto;
  margin-right:auto;
}

.projects-header .projects-header-footer p {
  margin: 0;
}

.projects-header .projects-header-footer .icon-btn {
  cursor: pointer;
  color: rgb(0, 225, 250);
  font-size: 30px;
}

.projects-header .projects-header-footer .icon-btn:hover {
  color: rgb(196, 63, 101);
}
