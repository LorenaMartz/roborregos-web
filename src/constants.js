export const LARGE_WIDTH = 1200
export const MEDIUM_WIDTH = 992
export const SMALL_WIDTH = 768
export const XSMALL_WIDTH = 592
export const MOBILE_WIDTH = 414
