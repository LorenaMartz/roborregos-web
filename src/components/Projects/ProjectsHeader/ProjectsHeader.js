// @flow
import React, { Component } from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faAngleDown } from '@fortawesome/free-solid-svg-icons'
import './ProjectsHeader.css'

type Props = {};

/** Component class of Projects' header. */
class ProjectsHeader extends Component<Props> {
  /**
 Function to scroll down the window view towards th end of the component.
 */
  scrollToInfo = () => {
    window.scrollBy(0, window.innerHeight - 48 - window.scrollY)
  }

  /**
 * Renders Responsive view of Projects's header.
 * @return {renderized_component} Heder banner with legend.
 */
  render() {
    return (
      <div className="projects-header">
        <div className="container-legend">
          <h2 className="title-text-banner">
            Projects
          </h2>
          <div className="main-text-projects">
            <p>
              Every year we challenge ourselves to keep learning and improve.
              See what we&apos;ve been up to!
            </p>
          </div>
        </div>
        <div className="projects-header-footer">
          <p>
            Scroll down and meet the us!
          </p>
          <FontAwesomeIcon
            onClick={this.scrollToInfo}
            icon={faAngleDown}
            className="icon-btn"
          />
        </div>
      </div>
    )
  }
}

export default ProjectsHeader
