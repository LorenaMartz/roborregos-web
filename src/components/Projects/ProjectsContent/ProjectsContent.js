// @flow
import React, { Component } from 'react'
import './ProjectsContent.css'
import ProjectsCard from './ProjectsCard/ProjectsCard'
import ProjectsCarousel from './ProjectsCarousel/ProjectsCarousel'
import ProjectsOther from './ProjectsOther/ProjectsOther'
import { SMALL_WIDTH } from '../../../constants'

type Project = {
  title: string,
  description: string,
  image: string,
  wiki: string
};

type OtherProjects = {
  title: string,
  image: string,
  wiki: string,
  background: string,
  color: string
};

type Props = {
  main_projects: Array<Project>,
  carousels: Array<Array<Project>>,
  other_projects: Array<OtherProjects>
};

type State = {
  visible: Array<boolean>
};

/** Component class of Projects page. */
class ProjectsContent extends Component <Props, State> {
  main_projects: Array<Project>;

  carousels: Array<Array<Project>>;

  other_projects: Array<OtherProjects>;

  /**
   * Class constructor
   * @param {list} props: List of projects by types.
   */
  constructor(props: Props) {
    super(props)

    this.main_projects = props.main_projects
    this.carousels = props.carousels
    this.other_projects = props.other_projects
    this.state = {
      visible: [],
    }
  }

  componentDidMount = () => {
    window.addEventListener('resize', this.updateView)
    window.addEventListener('scroll', this.makeAppear, true)
  }

  makeAppear = () => {
    this.setState({
      visible: this.updateVisibility(),
    })
  }

  listenScrollEvent = (card_index) => (window.innerHeight * (card_index) < window.scrollY)

  updateVisibility = () => {
    const visibility = []
    for (let x = 0; x < this.main_projects.length; x += 1) {
      visibility[x] = this.listenScrollEvent(x)
    }
    return visibility
  }

  viewSizeLarge = () => {
    if (window.innerWidth > SMALL_WIDTH) {
      return true
    }
    return false
  }

  /**
   * Renders Responsive view of Projects's body page.
   * @return {renderized_component} Heder, grid, join us section and footer.
   */
  render() {
    const { visible } = this.state
    return (
      <div className="projects-content">
        {
            this.main_projects.map((project, index) => {
              if (visible[index]) {
                return (
                  <div className="main-project-card-container">
                    <ProjectsCard
                      project={project}
                      index={index}
                      show_scrollers
                    />
                  </div>
                )
              }
              return (<div />)
            })
        }
        <ProjectsCarousel project_carousels={this.carousels} main_counter={this.main_projects.length} />
        <ProjectsOther projects={this.other_projects} />
      </div>
    )
  }
}

export default ProjectsContent
