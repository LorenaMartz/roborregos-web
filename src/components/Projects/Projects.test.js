// @flow
import React from 'react'
import { configure, mount, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import ProjectsComponent from './Projects'
import ProjectsHeader from './ProjectsHeader/ProjectsHeader'
import ProjectsContent from './ProjectsContent/ProjectsContent'
import Footer from '../Footer/Footer'
import { LARGE_WIDTH, SMALL_WIDTH } from '../../constants'

// Mock data
const projectsData = {
  main:
            [
              {
                title: 'main-project-1',
                description: 'description-main-1',
                wiki: 'https://github.com/',
                image: 'main-project-1',
              },
              {
                title: 'main-project-2',
                description: 'description-main-2',
                wiki: 'https://github.com/',
                image: 'main-project-2',
              },
            ],
  carrousel:
        [
          [
            {
              title: 'carousel1-project-1',
              description: 'description-carousel1-1',
              wiki: 'https://github.com/carousel1',
              image: 'carousel1-project-1',
            },
          ],
          [
            {
              title: 'carousel2-project-1',
              description: 'description-carousel2-1',
              wiki: 'https://github.com/carousel2',
              image: 'carousel2-project-1',
            },
          ],
        ],
  other:
            [
              {
                title: 'other_project1',
                wiki: 'https://github.com/1',
                image: 'other_project1',
                background: '#FFFFFF',
                color: 'black',
              },
            ],
}

let wrapper
configure({ adapter: new Adapter() })

beforeEach(() => {
  wrapper = mount(
    <ProjectsComponent projectsData={projectsData} />,
  )
})

it('<Projects> render main components correctly', () => {
  expect(wrapper.find(ProjectsComponent).length).toBe(1)
  expect(wrapper.find(ProjectsHeader).length).toBe(1)
  expect(wrapper.find(ProjectsContent).length).toBe(1)
  expect(wrapper.find(Footer).length).toBe(1)
})

// Snapshots
describe('<Projects> matches snapshot with:', () => {
  it('Large width', () => {
    global.innerWidth = LARGE_WIDTH
    global.dispatchEvent(new Event('resize'))
    const wrapper_content = shallow(
      <ProjectsComponent projectsData={projectsData} />,
    ).debug()
    expect(wrapper_content).toMatchSnapshot('ProjectsLargeView.test.js.snap')
  })

  it('Small width', () => {
    global.innerWidth = SMALL_WIDTH
    global.dispatchEvent(new Event('resize'))
    const wrapper_content = shallow(
      <ProjectsComponent projectsData={projectsData} />,
    ).debug()
    expect(wrapper_content).toMatchSnapshot('ProjectsSmallView.test.js.snap')
  })
})
