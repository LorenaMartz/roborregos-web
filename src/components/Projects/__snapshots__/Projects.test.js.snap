// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Projects> matches snapshot with: Large width: ProjectsLargeView.test.js.snap 1`] = `
"<div className=\\"projects-container\\">
  <HeaderBanner title=\\"Projects\\" mainText={{...}} subText={{...}} bgColorScheme={{...}} iconColorScheme={{...}} />
  <ProjectsContent main_projects={{...}} carousels={{...}} other_projects={{...}} />
  <Footer />
</div>"
`;

exports[`<Projects> matches snapshot with: Small width: ProjectsSmallView.test.js.snap 1`] = `
"<div className=\\"projects-container\\">
  <HeaderBanner title=\\"Projects\\" mainText={{...}} subText={{...}} bgColorScheme={{...}} iconColorScheme={{...}} />
  <ProjectsContent main_projects={{...}} carousels={{...}} other_projects={{...}} />
  <Footer />
</div>"
`;
